src/main/java/com/entropyteam/entropay/employees/services/EmployeeService.java

```diff
@@ -120,4 +123,45 @@ public List<String> getColumnsForSearch() {
         return Arrays.asList("firstName", "lastName", "internalId");
     }
 
+    public Integer applyVacationRuleToEmployee(Employee employee, String vacationYearToAdd, List<Contract> employeeContracts, List<Holiday> holidaysInPeriod) {
+        Optional<Contract> activeContract = employeeContracts.stream()
+                .filter(Contract::isActive)
+                .findFirst();
+        Optional<Contract> firstContract = employeeContracts.stream().min(Comparator.comparing(Contract::getStartDate));
+        boolean hasVacationsLoad = vacationRepository.existsVacationByEmployeeIdAndDeletedIsFalseAndYearIsLike(employee.getId(), vacationYearToAdd);
+
+        if (activeContract.isPresent() && firstContract.isPresent() && !hasVacationsLoad) {
+            int vacationDays = activeContract.get().getSeniority().getVacationDays();
+            LocalDate startDate = firstContract.orElse(null).getStartDate();
```

Comment:
we should change to `LocalDate startDate = firstContract.get().getStartDate();` because you already check for isPresent() before